pkg/http/handler/timeout.go

```diff
@@ -97,6 +112,9 @@ func (h *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				done <- p
 			}
 		}()
+		tw.mu.Lock()
+		tw.requestStartTime = tw.clock.Now()
+		tw.mu.Unlock()
```

Comment:
This seems unnecessary to do locking for. Can't you add the time above when constructing the `timeoutWriter`?

Reply:
Regarding the lock `requestStartTime` is also read by tw.tryMaxDurationTimeoutAndWriteError in: 
```		case cur := <-maxDurationTimeoutCh:
			timedOut := tw.tryMaxDurationTimeoutAndWriteError(cur, h.maxDurationTimeout, h.body)
			if timedOut {
				maxDurationTimeoutDrained = true
				return
			} \\\

so I guess to be on the safe side I should move it earlier above and remove the lock.